### PR TITLE
[Program: GCI] feat: add password confirmation functionality in POST /register

### LINKED ISSUE #252: Password Confirmation functionality for Registration
Whenever a user registers on the mentorship portal, he/she should enter the desired password and confirm the same before registering.

**Describe the solution you'd like**
Two password input fields should be present: password and confirm_password on POST/register API on the mentorship-backend

### PR DESCRIPTION
**Pull request by daksha-aeer**
### Description
Password confirmation functionality added.

**Working**
1. PASSWORD_CONFIRMATION_MISMATCH added in messages.py
2. confirm_password field added in app/api/models/user.py under register_user_api_model
3. password and confirm_password matched in app/api/dao/user.py . 400 response with message sent if they don't match
4. Updated test cases

Fixes #252 

### Type of Change:
- Code

**Code/Quality Assurance Only**
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
Screenshots
1. Successful registration
task18_3
task18_4

2. Unsuccessful registration (duplicate password)
task18_5
task18_6

3. Tests
```sh
python -m unittest discover tests
```
image

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas

**Code/Quality Assurance Only**
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by sara-02** [comment]
@daksha-aeer your tests are still failing. The existing /register test needs to be updated with the new `password_confirmation` field, else they will keep failing.

**Comment by daksha-aeer** [comment]
Ok adding new test

**Comment by sara-02** [comment]
Sure. We did not take this into consideration for the task but since we are modifying an existing feature that has tests against it, the tests will fail.

File: app/api/dao/user.py [hunk 1]
```diff
@@ -32,9 +32,11 @@ def create_user(data):
         name = data['name']
         username = data['username']
         password = data['password']
+        confirm_password = data['confirm_password']
         email = data['email']
         terms_and_conditions_checked = data['terms_and_conditions_checked']
-
+        if (password!=confirm_password):
```
Comment (sara-02) [inline app/api/dao/user.py]: space around `!=`, we don't need brackets around if condition.

File: app/messages.py [hunk 2]
```diff
@@ -213,3 +213,6 @@
 # Miscellaneous
 VALIDATION_ERROR = {"message": "Validation error."}
 NOT_IMPLEMENTED = {"message": "Not implemented."}
+
+# Password confirmation
+PASSWORD_CONFIRMATION_MISMATCH={"message": "The 2 password fields do not match."}
```
Comment (sara-02) [inline app/messages.py]: space around `=`.
`two` not 2

**Comment by daksha-aeer** [comment]
Is this travis thing related to those tests?

**Review by sara-02 — CHANGES_REQUESTED** [review]

**Comment by sara-02** [comment]
> Is this travis thing related to those tests?

Yes. If you click on `details` on the right side of the CI line, you can see the logs and check the error there as well.

**Comment by daksha-aeer** [comment]
I've updated the code, please review

File: tests/users/test_api_resources.py [hunk 3]
```diff
@@ -31,6 +31,7 @@ def test_user_registration_without_optional_fields(self, send_email_function):
                 name=user1['name'],
                 username=user1['username'],
                 password=user1['password'],
+                confirm_password=user1['password'],
```
Comment (sara-02) [inline tests/users/test_api_resources.py]: space around `=`

File: tests/users/test_api_resources.py [hunk 4]
```diff
@@ -55,6 +56,7 @@ def test_user_registration_with_both_optional_fields(self, send_email_function):
...
```
Comment (sara-02) [inline tests/users/test_api_resources.py]: same as above

File: tests/users/test_api_resources.py [hunk 5]
```diff
@@ -72,6 +74,7 @@ def test_user_registration_mentor(self, send_email_function):
...
```
Comment (sara-02) [inline tests/users/test_api_resources.py]: same as above

File: tests/users/test_api_resources.py [hunk 6]
```diff
@@ -88,6 +91,7 @@ def test_user_registration_mentee(self, send_email_function):
...
```
Comment (sara-02) [inline tests/users/test_api_resources.py]: same as above.

File: tests/users/test_dao.py [hunk 7]
```diff
@@ -23,6 +23,7 @@ def test_dao_create_user(self):
             username='user2',
             email='user2@email.com',
             password='test_password',
+            confirm_password='test_password',
```
Comment (sara-02) [inline tests/users/test_dao.py]: same as above.

**Review by sara-02 — CHANGES_REQUESTED** [review]

File: tests/users/test_dao.py (hunk 7 above)
Comment (daksha-aeer) [reply tests/users/test_dao.py]: done, please check

**Comment by daksha-aeer** [comment]
Hi @sara-02 I've made all changes, please look at them 😊

**Review by sara-02 — APPROVED** [review]

**Comment by isabelcosta** [comment]
Thank you for your contribution during GCI!
I will close this PR, as this feature is not currently desired [(comment)](https://github.com/anitab-org/mentorship-backend/issues/252#issuecomment-564300077).

**Pull request closed without merging**